### PR TITLE
cmake: remove include/drivers from include lists

### PR DESCRIPTION
**Pull request by pabigot**
Files that need something from this directory should be using the
prefix, e.g. <drivers/sensor.h> instead of just <sensor.h>.

Signed-off-by: Peter Bigot <peter.bigot@nordicsemi.no>

### CODE REVIEW

**Review by nashif — APPROVED** [review]

**Comment by pabigot** [comment]
Blocked by #21974.

**Review by carlescufi — APPROVED** [review]

**Comment by pabigot** [comment]
This got ugly.  I think this now fixes all the references that break when `drivers` is not an include root, but we'll see what CI says.  zephyrproject-rtos/mcumgr#10 must be merged and west updated before the mcumgr sample will build.

**Comment by pabigot** [comment]
Also note that because this has tree-wide changes the cleanup may have to be done again several times as things are merged in.  The script below automates the changes.
```
(cd include/drivers ;
 find . -name '*.h' \
   | sed -e 's@^\./@@' \
   | sed -e '\@console/console.h@d' \
   | sort
) | while read f ; do
git grep -l "<${f}>" | xargs sed -i -r -e "s@<${f}>@<drivers/${f}>@g"
git grep -l '"'${f}'"' | xargs sed -i -r -e 's@"'${f}'"@<drivers/'${f}'>@g'
(cat <<EOText
treewide: use full path to ${f} header

The build infrastructure should not be adding the drivers subdirectory
to the include path.  Fix the legacy uses that depended on that
addition.
EOText
) > /tmp/x
git commit -a -F/tmp/x -s || true
done
```

**Review by jhedberg — APPROVED** [review]

**Review by joerchan — APPROVED** [review]

**Review by vanwinkeljan — APPROVED** [review]

**Review by ydamigos — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Review by masz-nordic — APPROVED** [review]

**Review by finikorg — APPROVED** [review]

**Review by SebastianBoe — APPROVED** [review]

**Review by jfischer-no — APPROVED** [review]

**Review by alexanderwachter — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Review by pfalcon — APPROVED** [review]

**Comment by pabigot** [comment]
DNM until zephyrproject-rtos/mcumgr#10 is merged and the west commit in this PR updated.

**Comment by pabigot** [comment]
Blocked by #22128.

**Comment by pabigot** [comment]
Subsequent merges require a re-run of the conversion, and mcuboot also needs an update.  I'll do that this weekend, hopefully it'll be quiet enough to get a CI pass and merge quickly before more conflicts arise.

File: west.yml
Comment (nvlsianpu) [inline west.yml]: 54c1e3fb6ba4dcf7b315943bcfcda1e10a4042f5

**Review by nvlsianpu — CHANGES_REQUESTED** [review]

File: west.yml
Comment (pabigot) [reply west.yml]: done

**Review by nvlsianpu — APPROVED** [review]

**Comment by carlescufi** [comment]
@pabigot why is this marked as DNM?

**Comment by pabigot** [comment]
> @pabigot why is this marked as DNM?

Because it had not yet completed a successful CI after the final update of the mcuboot west module this morning; I wanted to make sure it was right.

It still hasn't, but the failure is due to fragile CI: the last test related to kernel.futex seems to never finish nor be cancelled.  So I've removed the label.

**Pull request merged**
